src/baseClient.ts

```diff
@@ -84,6 +84,35 @@ export class ClasschartsClient {
       }
     );
   }
+  
+  /**
+   * Returns all of the activities from and to the indicated dates.
+   * An abstraction on top of the getActivity method, it fetches everything in the given interval instead of returning a fragment.
+   */
+  async getFullActivity(from: string, to: string, client: ClasschartsClient) {
+    let data: ActivityResponse = [];
+    let prevLast: number | undefined;
+    
+    while (true) {
+      const params: GetActivityOptions = {
+        from,
+        to,
+      };
+      if (prevLast) {
+        params.last_id = `${prevLast}`;
+      }
+      const fragment = await client.getActivity(params);
```

Comment:
```suggestion
      const fragment = await this.getActivity(params);
```
We can use the existing client